{{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}plugin.py

```diff
@@ -10,56 +10,99 @@
+    from spyder.api.preferences import PluginConfigPage
 except ImportError:
     from spyder.plugins.configdialog import PluginConfigPage # Spyder3
     {%- endif %}
-{% else -%}
+{% else %}
+    {%- if cookiecutter.graphical_plugin == 'y' %}
 from spyder.api.plugins import SpyderPluginWidget
+    {%- else %}
+from spyder.api.plugins import SpyderPlugin
+    {%- endif %}
     {%- if cookiecutter.create_config_page == 'y' %}
 from spyder.api.preferences import PluginConfigPage
-    {%- endif %}
-{%- endif -%}
-
+    {% endif %}
+{%- endif %}
 from .widgets.{{ cookiecutter.plugin_name.lower().replace(' ', '') }}gui import {{ widget_name }}
-{% if cookiecutter.create_config_page == 'y' %}
 
+{% if cookiecutter.create_config_page == 'y' %}
 class {{ config_page }}(PluginConfigPage):
     """{{ cookiecutter.plugin_name }} plugin preferences."""
     pass
 {% endif %}
+{%- if cookiecutter.graphical_plugin == 'n' %}
+class {{ cookiecutter.plugin_name.replace(' ', '') }}Plugin(SpyderPlugin):
+    """{{ cookiecutter.plugin_name }} plugin."""
+    CONF_SECTION = '{{ cookiecutter.plugin_name }}'
+    {% if cookiecutter.create_config_page == 'y' %}
+    CONFIGWIDGET_CLASS = {{ config_page }}
+    {% else %}
+    CONFIGWIDGET_CLASS = None
+    {% endif %}
+    def __init__(self, parent=None):
+        SpyderPlugin.__init__(self, parent)
+
+        # Create widget and add to dockwindow
+        self.widget = {{ widget_name }}(self.main)
+
+        # Initialize plugin
+        self.initialize_plugin()
+
+    # --- SpyderPlugin API ----------------------------------------------
+    def get_plugin_title(self):
+        """Return widget title."""
+        return "{{ cookiecutter.plugin_name }}"
 
+    def refresh_plugin(self):
+        """Refresh {{ widget_name }} widget."""
+        pass
+
+    def get_plugin_actions(self):
+        """Return a list of actions related to plugin."""
+        return []
+
+    def register_plugin(self):
+        """Register plugin in Spyder's main window."""
+        pass
+
+    def on_first_registration(self):
+        """Action to be performed on first plugin registration."""
+        pass
+
+    def apply_plugin_settings(self, options):
+        """Apply configuration file's plugin settings."""
+        pass
+{%- else %}
 class {{ cookiecutter.plugin_name.replace(' ', '') }}Plugin(SpyderPluginWidget):
     """{{ cookiecutter.plugin_name }} plugin."""
     CONF_SECTION = '{{ cookiecutter.plugin_name }}'
-{% if cookiecutter.create_config_page == 'y' %}
+    {% if cookiecutter.create_config_page == 'y' %}
     CONFIGWIDGET_CLASS = {{ config_page }}
-{% else %}
+    {% else %}
     CONFIGWIDGET_CLASS = None
-{% endif %}
-
+    {% endif %}
     def __init__(self, parent=None):
         SpyderPluginWidget.__init__(self, parent)
-        self.main = parent # Spyder3
 
-        # Create widget and add to dockwindow
-        self.widget = {{ widget_name }}(self.main)
+        # Initialize plugin
+        self.initialize_plugin()
```

Comment:
This method is obsolete in Spyder 4, but it works in Spyder 3, so please take into account here.